src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -17,7 +17,111 @@ namespace System.Diagnostics.Tests
+        }
+
+        [Fact]
+        public void TestProcessOnRemoteMachineUnix()
+        {
+            Process currentProcess = Process.GetCurrentProcess();
+
+            Assert.Throws<PlatformNotSupportedException>(() => Process.GetProcessesByName(currentProcess.ProcessName, "127.0.0.1"));
+            Assert.Throws<PlatformNotSupportedException>(() => Process.GetProcessById(currentProcess.Id, "127.0.0.1"));
+        }
+
+        [Theory]
+        [MemberData(nameof(MachineName_Remote_TestData))]
+        public void GetProcessesByName_RemoteMachineNameUnix_ThrowsPlatformNotSupportedException(string machineName)
+        {
+            Process currentProcess = Process.GetCurrentProcess();
+            Assert.Throws<PlatformNotSupportedException>(() => Process.GetProcessesByName(currentProcess.ProcessName, machineName));
+        }
+
+        [Fact]
+        public void TestRootGetProcessById()
+        {
+            Process p = Process.GetProcessById(1);
+            Assert.Equal(1, p.Id);
+        }
+
+        [Fact]
+        public void TestUseShellExecute_Unix_Succeeds()
+        {
+            using (var p = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = "exit", Arguments = "42" }))
+            {
+                Assert.True(p.WaitForExit(WaitInMS));
+                Assert.Equal(42, p.ExitCode);
+            }
+        }
+
+        [Fact]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
+        public void TestPriorityClassUnix()
+        {
+            CreateDefaultProcess();
+
+            ProcessPriorityClass priorityClass = _process.PriorityClass;
+            try
+            {
+                _process.PriorityClass = ProcessPriorityClass.High;
+                Assert.Equal(_process.PriorityClass, ProcessPriorityClass.High);
+
+                _process.PriorityClass = ProcessPriorityClass.Normal;
+                Assert.Equal(_process.PriorityClass, ProcessPriorityClass.Normal);
+            }
+            catch (Win32Exception ex)
+            {
+                Assert.True(!PlatformDetection.IsSuperUser, $"Failed even though superuser {ex.ToString()}");
+            }
+            finally
+            {
+                _process.PriorityClass = priorityClass;
+            }
+        }
+
+        [Fact]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
+        public void TestBasePriorityOnUnix()
+        {
+            CreateDefaultProcess();
+
+            ProcessPriorityClass originalPriority = _process.PriorityClass;
+            Assert.Equal(ProcessPriorityClass.Normal, originalPriority);
+
+            try
+            {
+                SetAndCheckBasePriority(ProcessPriorityClass.High, -11);
+                SetAndCheckBasePriority(ProcessPriorityClass.Idle, 19);
+                SetAndCheckBasePriority(ProcessPriorityClass.Normal, 0);
+            }
+            catch (Win32Exception ex)
```

Comment:
I didn't filter out the test entirely because this means we check that unelevated throws Win32Exception and not something else.

I went ahead and pulled out the lines that didn't need elevation so they run either way.